benchmark/async_hooks/promises.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+const common = require('../common.js');
+
+const bench = common.createBenchmark(main, {
+  n: [1e6],
+  method: [
+    'trackingEnabled',
+    'trackingDisabled',
+  ]
+});
+
+async function run(n) {
+  for (let i = 0; i < n; i++) {
+    await new Promise((resolve) => resolve())
+    .then(() => { throw new Error('foobar'); })
+    .catch((e) => e);
+  }
+}
+
+function main({ n, method }) {
+  const hook = require('async_hooks').createHook({ promiseResolve() {} });
+  switch (method) {
+    case 'trackingEnabled':
+      hook.enable();
```

Comment:
Nit: I'd prefer for hook require to go at the top of the file and the `enable()` in the respective branch.